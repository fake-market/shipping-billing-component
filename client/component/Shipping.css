label {
  font-size: 12px;
  font-family: helvetica, arial, sans-serif;
}

select {
  font-size: 14px;
  font-family: helvetica, arial, sans-serif;
}

input {
  font-size: 14px;
  font-family: helvetica, arial, sans-serif;
}

button {
  border-radius: 4px;
  color: blue;
  font-size: 15px;
  font-weight: 600;
}
