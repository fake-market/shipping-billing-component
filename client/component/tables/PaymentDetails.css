:local(.payment) {
  border: 1px solid black;
  font-size: 18px;
  font-family: helvetica, arial, sans-serif;
  width: 100%;
}

th {
  height: 40px;
}

td {
  height: 30px;
}
