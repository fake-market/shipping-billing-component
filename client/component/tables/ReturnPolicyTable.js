import React from 'react';

const ReturnPolicyTable = () => (
  <div></div>
);

export default ReturnPolicyTable;
