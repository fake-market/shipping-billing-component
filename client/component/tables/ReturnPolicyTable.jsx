import React from 'react';

import styles from './ReturnPolicyTable.css';

const ReturnPolicyTable = () => (
  <div>
    <table className={styles.return}>
      <h3>Return Policy</h3>
      <tr>
        <th>After receiving item, contact seller within</th>
        <th>Refund will be given as</th>
        <th>Return shipping</th>
      </tr>
      <tr>
        <td>30 days</td>
        <td>Money back</td>
        <td>Seller pays for return shipping</td>
      </tr>
      <tr>
        <td colspan="3">
          <p>
              Refer to <a href="https://www.ebay.com/help/buying/returns-refunds/return-item-refund?id=4041#section3">eBay Return policy</a>
            for more details. You are covered by the eBay Money Back Guarantee if you receive an item that is not as described in the listing.
          </p>
        </td>
      </tr>
      <tr>
        <th colspan="3">Return policy details</th>
      </tr>
      <tr>
        <td colspan="3">
          All buyers remorse returns or returns for refused delivery will be subject to 20% restocking fee.
          Any items that are returned not complete or with damages pieces will not be refunded.
        </td>
      </tr>
    </table>
  </div>
);

export default ReturnPolicyTable;
