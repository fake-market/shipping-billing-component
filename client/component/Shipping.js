import React from 'react';
import axios from 'axios';

import { seeds } from '../../db/sql/seed';

import ShippingTable from './tables/ShippingTable';
import ReturnPolicyTable from './tables/ReturnPolicyTable';

class Shipping extends React.Component {
  constructor() {
    super();

    const countries = seeds.map(item => item.country);

    this.state = {
      defaultCountry: 'United States of America',
      selectedCountry: '',
      countries: [...countries],
      basicRate: '',
      expeditedRate: '',
      oneDayRate: '',
      quantity: 1
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    // not implemented yet:
    // this.handleChangeInQty = this.handleChangeInQty.bind(this);
  }

  componentDidMount() {
    this.handleDefault();
  }

  handleDefault() {
    axios({
      method: 'get',
      url: 'api/shipping',
      params: {
        country: this.state.defaultCountry
      }
    })
    .then(({ data }) => {
      console.log(data);

      this.setState({
        basicRate: data.basic_rate,
        expeditedRate: data.expedited_rate,
        oneDayRate: data.one_day_rate
      });
    })
    .catch(err => {
      console.log('error fetching from client');
    });
  }

  handleSubmit(event) {
    event.preventDefault();

    if (this.state.selectedCountry) {
      axios({
        method: 'get',
        url: 'api/shipping',
        params: {
          country: this.state.selectedCountry
        }
      })
      .then(({ data }) => {
        this.setState({
          defaultCountry: data.country,
          basicRate: data.basic_rate,
          expeditedRate: data.expedited_rate,
          oneDayRate: data.one_day_rate
        });
      })
      .catch(err => {
        console.log('error fetching from client');
      });
    }
  }

  handleChange(event) {
    this.setState({
      selectedCountry: event.target.value
    }, () => console.log(this.state.selectedCountry));
  }

  //not implemented on the actual ebay site
  /*
  handleChangeInQty(event) {
    console.log(event.target.value);
    let expeditedFee = this.state.quantity * 3;
    let oneDayRateFee = this.state.quantity * 5;

    if (this.state.quantity > 1) {
      let totalExp = this.state.expeditedRate + expeditedFee;
      let totalOne = this.state.oneDayRate + oneDayRateFee;

      this.setState({
        expeditedRate: totalExp,
        oneDayRate: totalOne
      });
    }
  }
  *///////////////////////////////////////////

  render() {
    let tableStyle = {
      width: '100%'
    };

    return(
      <div>
        <h1>Shipping and Billing</h1>
        <form
          onSubmit={this.handleSubmit}
        >
          {"QTY: "}<input
            onChange={this.handleChangeInQty}
            type="number" min="1"
          ></input>{" "}
          <label>
            { "Change Country: " }
            <select
              value={this.state.selectedCountry || this.state.defaultCountry}
              onChange={this.handleChange}
              >
              {
                this.state.countries.map(country => (
                  <option>{country}</option>
                ))
              }
            </select>
          </label>
          {" "}<button>Get Rates</button>
        </form>
        <br></br>
        <div>
          <ShippingTable
            basicRate={this.state.basicRate}
            expeditedRate={this.state.expeditedRate}
            oneDayRate={this.state.oneDayRate}
            defaultCountry={this.state.defaultCountry}
          />
        </div>
        <div>
          <ReturnPolicyTable />
        </div>
      </div>
    )
  }
}

export default Shipping;
