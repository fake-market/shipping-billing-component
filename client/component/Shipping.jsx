import React from 'react';
import axios from 'axios';

import { seeds } from '../../db/sql/seed';

import styles from './Shipping.css';

import ReturnPolicyTable from './tables/ReturnPolicyTable.jsx';
import PaymentDetails from './tables/PaymentDetailsTable.jsx';

class Shipping extends React.Component {
  constructor() {
    super();

    const countries = seeds.map(item => item.country);

    this.state = {
      defaultCountry: 'United States of America',
      selectedCountry: '',
      countries: [...countries],
      basicRate: '',
      expeditedRate: '',
      oneDayRate: '',
      quantity: 1
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    // not implemented yet:
    // this.handleChangeInQty = this.handleChangeInQty.bind(this);
  }

  componentDidMount() {
    this.handleDefault();
  }

  handleDefault() {
    axios({
      method: 'get',
      url: 'api/shipping',
      params: {
        country: this.state.defaultCountry
      }
    })
    .then(({ data }) => {
      console.log(data);

      this.setState({
        basicRate: data.basic_rate,
        expeditedRate: data.expedited_rate,
        oneDayRate: data.one_day_rate
      });
    })
    .catch(err => {
      console.log('error fetching from client');
    });
  }

  handleSubmit(event) {
    event.preventDefault();

    if (this.state.selectedCountry) {
      axios({
        method: 'get',
        url: 'api/shipping',
        params: {
          country: this.state.selectedCountry
        }
      })
      .then(({ data }) => {
        this.setState({
          defaultCountry: data.country,
          basicRate: data.basic_rate,
          expeditedRate: data.expedited_rate,
          oneDayRate: data.one_day_rate
        });
      })
      .catch(err => {
        console.log('error fetching from client');
      });
    }
  }

  handleChange(event) {
    this.setState({
      selectedCountry: event.target.value
    }, () => console.log(this.state.selectedCountry));
  }

  //not implemented on the actual ebay site
  /*
  handleChangeInQty(event) {
    console.log(event.target.value);
    let expeditedFee = this.state.quantity * 3;
    let oneDayRateFee = this.state.quantity * 5;

    if (this.state.quantity > 1) {
      let totalExp = this.state.expeditedRate + expeditedFee;
      let totalOne = this.state.oneDayRate + oneDayRateFee;

      this.setState({
        expeditedRate: totalExp,
        oneDayRate: totalOne
      });
    }
  }
  *///////////////////////////////////////////

  render() {
    return(
      <div>
        <div className={styles['shipping-outside-border']}>
          <h3>Shipping and Billing</h3>
          <div className={styles['shipping-conditions']}>
            <div>Item location: <b>Windsor, New Jersey, United States</b></div>
            <div>Shipping to: Worldwide</div>
            <div>Excludes: Russian Federation</div>
          </div>
          <form
            className={styles['form-padding']}
            onSubmit={this.handleSubmit}
          >
            <label>{ "Quantity: "}</label>
            <input
              onChange={this.handleChangeInQty}
              type="number" min="1"
            ></input>{" "}
            <label>
              { "Change Country: " }
              <select
                value={this.state.selectedCountry || this.state.defaultCountry}
                onChange={this.handleChange}
              >
                {
                  this.state.countries.map((country, i) => (
                    <option key={i}>{country}</option>
                  ))
                }
              </select>
            </label>
            {" "}<button>Get Rates</button>
          </form>
          <br></br>
          <div>
            <table className={styles.shipping}>
              <thead>
                <tr className={styles['shipping-header']}>
                  <th>Shipping and handling</th>
                  <th>Each additional item</th>
                  <th>To</th>
                  <th>Service</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>{`US $${this.state.basicRate}.00`}</td>
                  <td>free</td>
                  <td>{this.state.defaultCountry}</td>
                  <td>Standard Shipping</td>
                </tr>
                <tr>
                  <td>{`US $${this.state.expeditedRate}.00`}</td>
                  <td>3</td>
                  <td>{this.state.defaultCountry}</td>
                  <td>Expedited Shipping</td>
                </tr>
                <tr>
                  <td>{`US $${this.state.oneDayRate}.00`}</td>
                  <td>5</td>
                  <td>{this.state.defaultCountry}</td>
                  <td>{`One-day Shipping`}</td>
                </tr>
              </tbody>
            </table>
          </div>
          <div className={styles['shipping-delivery-conditions']}>
            <div>
              * <a href="https://pages.ebay.com/help/buy/contextual/estimated-delivery.html">
                  Estimated delivery dates
                </a>{' '}
                include seller's handling time, origin ZIP Code, destination ZIP Code
                and time of acceptance and will depend on shipping service selected and receipt of
                {' '}<a href="https://pages.ebay.com/help/buy/contextual/domestic-handling-time.html">cleared payment</a>.{' '}
                Delivery times may vary, especially during peak periods.
            </div>
          </div>
        </div>
        <br></br>
        <div>
          <ReturnPolicyTable />
        </div>
        <br></br>
        <div>
          <PaymentDetails />
        </div>
      </div>
    );
  }
}

export default Shipping;
