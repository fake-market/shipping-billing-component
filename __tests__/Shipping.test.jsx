import React from 'react';
import Enzyme, {shallow, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

import sinon from 'sinon';

import styles from '../client/component/Shipping.css';

import Shipping from '../client/component/Shipping';
import ReturnPolicyTable from '../client/component/tables/ReturnPolicyTable';
import PaymentDetails from '../client/component/tables/PaymentDetailsTable';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

configure({ adapter: new Adapter() });

////////////////////////////////////////////////////////////////////////////////
// test starts here: ///////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////

describe('Component: App', () => {
  const wrapper = shallow(<Shipping />)
  const countries = ['Philippines', 'Japan', 'United States of America'];

  beforeEach(() => {
    wrapper.setState({countries: countries});
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
  });

  it('Shipping should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('Shipping should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Shipping should render all nodes', () => {
    expect(wrapper.find('div').length).toBe(11);
  });

  it('Shipping should render options', () => {
    const instance = wrapper.instance();

    expect(instance.state.countries.length).toEqual(3);
  });

  it('Shipping should render ReturnPolicyTable', () => {
    expect(wrapper.contains(<ReturnPolicyTable />)).toBe(true);
    // expect(wrapper.find(ReturnPolicyTable).length).toBe(1);
  });

  it('Shipping should render PaymentDetailsTable', () => {
    expect(wrapper.contains(<PaymentDetails />)).toBe(true);
    // expect(wrapper.find(PaymentDetails).length).toBe(1);
  });

  it('Shipping should render table', () => {
    // expect(wrapper.find('.styles.shipping')).toBeDefined();
    expect(wrapper.find('table').length).toBe(1);
  });

  it('Shipping should render form', () => {
    // expect(wrapper.find('.styles.form-padding')).toBeDefined();
    expect(wrapper.find('form').length).toBe(1);
  });


  it('Shipping should register a click', async() => {
    const onSubmitForm = jest.fn();
    const preventDefault = jest.fn();
    const wrapper = mount(<Shipping />);
    const instance = wrapper.instance();

    expect.assertions(1);

    wrapper
      .find('input')
      .simulate('change', {
        target: {
          value: 'India'
        }
      });

    wrapper
      .find('form')
      .simulate('submit', {preventDefault});

    // expect(preventDefault).toBeCalled();
    expect(onSubmitForm).toBeCalledWith('India');
  });
});

// test('User fetched name should be Leanne Graham', async() => {
//   expect.assertions(1);
//   const data = await functions.fetchUser();
//   expect(data.name).toEqual('Leanne Graham');
// });
