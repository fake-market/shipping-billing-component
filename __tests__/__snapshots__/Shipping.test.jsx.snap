// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App Shipping should register submit 1`] = `
<div>
  <div>
    <h3>
      Shipping and Billing
    </h3>
    <div>
      <div>
        Item location: 
        <b>
          Windsor, New Jersey, United States
        </b>
      </div>
      <div>
        Shipping to: Worldwide
      </div>
      <div>
        Excludes: Russian Federation
      </div>
    </div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Quantity: 
      </label>
      <input
        min="1"
        type="number"
      />
       
      <label>
        Change Country: 
        <select
          onChange={[Function]}
          value="United States of America"
        >
          <option
            key="0"
          >
            Afghanistan
          </option>
          <option
            key="1"
          >
            Albania
          </option>
          <option
            key="2"
          >
            Algeria
          </option>
          <option
            key="3"
          >
            Andorra
          </option>
          <option
            key="4"
          >
            Angola
          </option>
          <option
            key="5"
          >
            Antigua and Barbud
          </option>
          <option
            key="6"
          >
            Argentina
          </option>
          <option
            key="7"
          >
            Armenia
          </option>
          <option
            key="8"
          >
            Australia
          </option>
          <option
            key="9"
          >
            Austria
          </option>
          <option
            key="10"
          >
            Azerbaijan
          </option>
          <option
            key="11"
          >
            Bahamas
          </option>
          <option
            key="12"
          >
            Bahrain
          </option>
          <option
            key="13"
          >
            Bangladesh
          </option>
          <option
            key="14"
          >
            Barbados
          </option>
          <option
            key="15"
          >
            Belarus
          </option>
          <option
            key="16"
          >
            Belgium
          </option>
          <option
            key="17"
          >
            Belize
          </option>
          <option
            key="18"
          >
            Benin
          </option>
          <option
            key="19"
          >
            Bhutan
          </option>
          <option
            key="20"
          >
            Bolivia
          </option>
          <option
            key="21"
          >
            Bosnia and Herzegovina
          </option>
          <option
            key="22"
          >
            Botswana
          </option>
          <option
            key="23"
          >
            Brazil
          </option>
          <option
            key="24"
          >
            Brunei
          </option>
          <option
            key="25"
          >
            Bulgaria
          </option>
          <option
            key="26"
          >
            Burkina Faso
          </option>
          <option
            key="27"
          >
            Burundi
          </option>
          <option
            key="28"
          >
            CÃ´te d'Ivoire
          </option>
          <option
            key="29"
          >
            Cabo Verde
          </option>
          <option
            key="30"
          >
            Cambodia
          </option>
          <option
            key="31"
          >
            Cameroon
          </option>
          <option
            key="32"
          >
            Canada
          </option>
          <option
            key="33"
          >
            Central African Republic
          </option>
          <option
            key="34"
          >
            Chad
          </option>
          <option
            key="35"
          >
            Chile
          </option>
          <option
            key="36"
          >
            China
          </option>
          <option
            key="37"
          >
            Colombia
          </option>
          <option
            key="38"
          >
            Comoros
          </option>
          <option
            key="39"
          >
            Congo (Congo-Brazzaville)
          </option>
          <option
            key="40"
          >
            Costa Rica
          </option>
          <option
            key="41"
          >
            Croatia
          </option>
          <option
            key="42"
          >
            Cuba
          </option>
          <option
            key="43"
          >
            Cyprus
          </option>
          <option
            key="44"
          >
            Czech Republic
          </option>
          <option
            key="45"
          >
            Democratic Republic of the Congo
          </option>
          <option
            key="46"
          >
            Denmark
          </option>
          <option
            key="47"
          >
            Djibouti
          </option>
          <option
            key="48"
          >
            Dominica
          </option>
          <option
            key="49"
          >
            Dominican Republic
          </option>
          <option
            key="50"
          >
            Ecuador
          </option>
          <option
            key="51"
          >
            Egypt
          </option>
          <option
            key="52"
          >
            El Salvador
          </option>
          <option
            key="53"
          >
            Equatorial Guinea
          </option>
          <option
            key="54"
          >
            Eritrea
          </option>
          <option
            key="55"
          >
            Estonia
          </option>
          <option
            key="56"
          >
            Ethiopia
          </option>
          <option
            key="57"
          >
            Fiji
          </option>
          <option
            key="58"
          >
            Finland
          </option>
          <option
            key="59"
          >
            France
          </option>
          <option
            key="60"
          >
            Gabon
          </option>
          <option
            key="61"
          >
            Gambia
          </option>
          <option
            key="62"
          >
            Georgia
          </option>
          <option
            key="63"
          >
            Germany
          </option>
          <option
            key="64"
          >
            Ghana
          </option>
          <option
            key="65"
          >
            Greece
          </option>
          <option
            key="66"
          >
            Grenada
          </option>
          <option
            key="67"
          >
            Guatemala
          </option>
          <option
            key="68"
          >
            Guinea
          </option>
          <option
            key="69"
          >
            Guinea-Bissau
          </option>
          <option
            key="70"
          >
            Guyana
          </option>
          <option
            key="71"
          >
            Haiti
          </option>
          <option
            key="72"
          >
            Holy See
          </option>
          <option
            key="73"
          >
            Honduras
          </option>
          <option
            key="74"
          >
            Hungary
          </option>
          <option
            key="75"
          >
            Iceland
          </option>
          <option
            key="76"
          >
            India
          </option>
          <option
            key="77"
          >
            Indonesia
          </option>
          <option
            key="78"
          >
            Iran
          </option>
          <option
            key="79"
          >
            Iraq
          </option>
          <option
            key="80"
          >
            Ireland
          </option>
          <option
            key="81"
          >
            Israel
          </option>
          <option
            key="82"
          >
            Italy
          </option>
          <option
            key="83"
          >
            Jamaica
          </option>
          <option
            key="84"
          >
            Japan
          </option>
          <option
            key="85"
          >
            Jordan
          </option>
          <option
            key="86"
          >
            Kazakhstan
          </option>
          <option
            key="87"
          >
            Kenya
          </option>
          <option
            key="88"
          >
            Kiribati
          </option>
          <option
            key="89"
          >
            Kuwait
          </option>
          <option
            key="90"
          >
            Kyrgyzstan
          </option>
          <option
            key="91"
          >
            Laos
          </option>
          <option
            key="92"
          >
            Latvia
          </option>
          <option
            key="93"
          >
            Lebanon
          </option>
          <option
            key="94"
          >
            Lesotho
          </option>
          <option
            key="95"
          >
            Liberia
          </option>
          <option
            key="96"
          >
            Libya
          </option>
          <option
            key="97"
          >
            Liechtenstein
          </option>
          <option
            key="98"
          >
            Lithuania
          </option>
          <option
            key="99"
          >
            Luxembourg
          </option>
          <option
            key="100"
          >
            Macedonia (FYROM)
          </option>
          <option
            key="101"
          >
            Madagascar
          </option>
          <option
            key="102"
          >
            Malawi
          </option>
          <option
            key="103"
          >
            Malaysia
          </option>
          <option
            key="104"
          >
            Maldives
          </option>
          <option
            key="105"
          >
            Mali
          </option>
          <option
            key="106"
          >
            Malta
          </option>
          <option
            key="107"
          >
            Marshall Islands
          </option>
          <option
            key="108"
          >
            Mauritania
          </option>
          <option
            key="109"
          >
            Mauritius
          </option>
          <option
            key="110"
          >
            Mexico
          </option>
          <option
            key="111"
          >
            Micronesia
          </option>
          <option
            key="112"
          >
            Moldova
          </option>
          <option
            key="113"
          >
            Monaco
          </option>
          <option
            key="114"
          >
            Mongolia
          </option>
          <option
            key="115"
          >
            Montenegro
          </option>
          <option
            key="116"
          >
            Morocco
          </option>
          <option
            key="117"
          >
            Mozambique
          </option>
          <option
            key="118"
          >
            Myanmar (formerly Burma)
          </option>
          <option
            key="119"
          >
            Namibia
          </option>
          <option
            key="120"
          >
            Nauru
          </option>
          <option
            key="121"
          >
            Nepal
          </option>
          <option
            key="122"
          >
            Netherlands
          </option>
          <option
            key="123"
          >
            New Zealand
          </option>
          <option
            key="124"
          >
            Nicaragua
          </option>
          <option
            key="125"
          >
            Niger
          </option>
          <option
            key="126"
          >
            Nigeria
          </option>
          <option
            key="127"
          >
            North Korea
          </option>
          <option
            key="128"
          >
            Norway
          </option>
          <option
            key="129"
          >
            Oman
          </option>
          <option
            key="130"
          >
            Pakistan
          </option>
          <option
            key="131"
          >
            Palau
          </option>
          <option
            key="132"
          >
            Palestine State
          </option>
          <option
            key="133"
          >
            Panama
          </option>
          <option
            key="134"
          >
            Papua New Guinea
          </option>
          <option
            key="135"
          >
            Paraguay
          </option>
          <option
            key="136"
          >
            Peru
          </option>
          <option
            key="137"
          >
            Philippines
          </option>
          <option
            key="138"
          >
            Poland
          </option>
          <option
            key="139"
          >
            Portugal
          </option>
          <option
            key="140"
          >
            Qatar
          </option>
          <option
            key="141"
          >
            Romania
          </option>
          <option
            key="142"
          >
            Russia
          </option>
          <option
            key="143"
          >
            Rwanda
          </option>
          <option
            key="144"
          >
            Saint Kitts and Nevis
          </option>
          <option
            key="145"
          >
            Saint Lucia
          </option>
          <option
            key="146"
          >
            Saint Vincent and the Grenadines
          </option>
          <option
            key="147"
          >
            Samoa
          </option>
          <option
            key="148"
          >
            San Marino
          </option>
          <option
            key="149"
          >
            Sao Tome and Principe
          </option>
          <option
            key="150"
          >
            Saudi Arabia
          </option>
          <option
            key="151"
          >
            Senegal
          </option>
          <option
            key="152"
          >
            Serbia
          </option>
          <option
            key="153"
          >
            Seychelles
          </option>
          <option
            key="154"
          >
            Sierra Leone
          </option>
          <option
            key="155"
          >
            Singapore
          </option>
          <option
            key="156"
          >
            Slovakia
          </option>
          <option
            key="157"
          >
            Slovenia
          </option>
          <option
            key="158"
          >
            Solomon Islands
          </option>
          <option
            key="159"
          >
            Somalia
          </option>
          <option
            key="160"
          >
            South Africa
          </option>
          <option
            key="161"
          >
            South Korea
          </option>
          <option
            key="162"
          >
            South Sudan
          </option>
          <option
            key="163"
          >
            Spain
          </option>
          <option
            key="164"
          >
            Sri Lanka
          </option>
          <option
            key="165"
          >
            Sudan
          </option>
          <option
            key="166"
          >
            Suriname
          </option>
          <option
            key="167"
          >
            Swaziland
          </option>
          <option
            key="168"
          >
            Sweden
          </option>
          <option
            key="169"
          >
            Switzerland
          </option>
          <option
            key="170"
          >
            Syria
          </option>
          <option
            key="171"
          >
            Tajikistan
          </option>
          <option
            key="172"
          >
            Tanzania
          </option>
          <option
            key="173"
          >
            Thailand
          </option>
          <option
            key="174"
          >
            Timor-Leste
          </option>
          <option
            key="175"
          >
            Togo
          </option>
          <option
            key="176"
          >
            Tonga
          </option>
          <option
            key="177"
          >
            Trinidad and Tobago
          </option>
          <option
            key="178"
          >
            Tunisia
          </option>
          <option
            key="179"
          >
            Turkey
          </option>
          <option
            key="180"
          >
            Turkmenistan
          </option>
          <option
            key="181"
          >
            Tuvalu
          </option>
          <option
            key="182"
          >
            Uganda
          </option>
          <option
            key="183"
          >
            Ukraine
          </option>
          <option
            key="184"
          >
            United Arab Emirates
          </option>
          <option
            key="185"
          >
            United Kingdom
          </option>
          <option
            key="186"
          >
            United States of America
          </option>
          <option
            key="187"
          >
            Uruguay
          </option>
          <option
            key="188"
          >
            Uzbekistan
          </option>
          <option
            key="189"
          >
            Vanuatu
          </option>
          <option
            key="190"
          >
            Venezuela
          </option>
          <option
            key="191"
          >
            Viet Nam
          </option>
          <option
            key="192"
          >
            Yemen
          </option>
          <option
            key="193"
          >
            Zambia
          </option>
          <option
            key="194"
          >
            Zimbabwe
          </option>
        </select>
      </label>
       
      <button>
        Get Rates
      </button>
    </form>
    <br />
    <div>
      <table>
        <thead>
          <tr>
            <th>
              Shipping and handling
            </th>
            <th>
              Each additional item
            </th>
            <th>
              To
            </th>
            <th>
              Service
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              US $.00
            </td>
            <td>
              free
            </td>
            <td>
              United States of America
            </td>
            <td>
              Standard Shipping
            </td>
          </tr>
          <tr>
            <td>
              US $.00
            </td>
            <td>
              3
            </td>
            <td>
              United States of America
            </td>
            <td>
              Expedited Shipping
            </td>
          </tr>
          <tr>
            <td>
              US $.00
            </td>
            <td>
              5
            </td>
            <td>
              United States of America
            </td>
            <td>
              One-day Shipping
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div>
      <div>
        * 
        <a
          href="https://pages.ebay.com/help/buy/contextual/estimated-delivery.html"
        >
          Estimated delivery dates
        </a>
         
        include seller's handling time, origin ZIP Code, destination ZIP Code and time of acceptance and will depend on shipping service selected and receipt of
         
        <a
          href="https://pages.ebay.com/help/buy/contextual/domestic-handling-time.html"
        >
          cleared payment
        </a>
        .
         
        Delivery times may vary, especially during peak periods.
      </div>
    </div>
  </div>
  <br />
  <div>
    <div>
      <h3>
        Return Policy
      </h3>
      <table>
        <thead>
          <tr>
            <th>
              After receiving item, contact seller within
            </th>
            <th>
              Refund will be given as
            </th>
            <th>
              Return shipping
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              30 days
            </td>
            <td>
              Money back
            </td>
            <td>
              Seller pays for return shipping
            </td>
          </tr>
          <tr>
            <td
              colSpan="3"
            >
              <p>
                Refer to 
                <a
                  href="https://www.ebay.com/help/buying/returns-refunds/return-item-refund?id=4041#section3"
                >
                  eBay Return policy
                </a>
                 for more details. You are covered by the 
                <a
                  href="https://www.ebay.com/help/policies/ebay-money-back-guarantee-policy/ebay-money-back-guarantee-policy?id=4210"
                >
                  eBay Money Back Guarantee
                </a>
                 if you receive an item that is not as described in the listing.
              </p>
            </td>
          </tr>
          <tr>
            <th
              colSpan="3"
            >
              Return policy details
            </th>
          </tr>
          <tr>
            <td
              colSpan="3"
            >
              All buyers remorse returns or returns for refused delivery will be subject to 20% restocking fee. Any items that are returned not complete or with damages pieces will not be refunded.
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <br />
  <div>
    <div>
      <h3>
        Payment details
      </h3>
      <table>
        <thead>
          <tr>
            <th>
              Payment Method
            </th>
            <th>
              Preferred / Accepted
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <img
                src="test-file-stub"
              />
            </td>
            <td>
              PayPal Preferred
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Component: App Shipping should render correctly 1`] = `
<div>
  <div>
    <h3>
      Shipping and Billing
    </h3>
    <div>
      <div>
        Item location: 
        <b>
          Windsor, New Jersey, United States
        </b>
      </div>
      <div>
        Shipping to: Worldwide
      </div>
      <div>
        Excludes: Russian Federation
      </div>
    </div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Quantity: 
      </label>
      <input
        min="1"
        type="number"
      />
       
      <label>
        Change Country: 
        <select
          onChange={[Function]}
          value="United States of America"
        >
          <option
            key="0"
          >
            Philippines
          </option>
          <option
            key="1"
          >
            Japan
          </option>
          <option
            key="2"
          >
            United States of America
          </option>
        </select>
      </label>
       
      <button>
        Get Rates
      </button>
    </form>
    <br />
    <div>
      <table>
        <thead>
          <tr>
            <th>
              Shipping and handling
            </th>
            <th>
              Each additional item
            </th>
            <th>
              To
            </th>
            <th>
              Service
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              US $.00
            </td>
            <td>
              free
            </td>
            <td>
              United States of America
            </td>
            <td>
              Standard Shipping
            </td>
          </tr>
          <tr>
            <td>
              US $.00
            </td>
            <td>
              3
            </td>
            <td>
              United States of America
            </td>
            <td>
              Expedited Shipping
            </td>
          </tr>
          <tr>
            <td>
              US $.00
            </td>
            <td>
              5
            </td>
            <td>
              United States of America
            </td>
            <td>
              One-day Shipping
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div>
      <div>
        * 
        <a
          href="https://pages.ebay.com/help/buy/contextual/estimated-delivery.html"
        >
          Estimated delivery dates
        </a>
         
        include seller's handling time, origin ZIP Code, destination ZIP Code and time of acceptance and will depend on shipping service selected and receipt of
         
        <a
          href="https://pages.ebay.com/help/buy/contextual/domestic-handling-time.html"
        >
          cleared payment
        </a>
        .
         
        Delivery times may vary, especially during peak periods.
      </div>
    </div>
  </div>
  <br />
  <div>
    <ReturnPolicyTable />
  </div>
  <br />
  <div>
    <PaymentDetails />
  </div>
</div>
`;
